### PR TITLE
Annotate ProcessTree with @CheckForNull/@Nonnull

### PR DESCRIPTION
**Pull request by slonopotamus**
### Proposed changelog entries

* Internal: `ProcessTree` public API is now annotated with @Nullable/@Nonnull annotations

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Please use `@CheckForNull` instead of `Nullable`. CheckforNull will require static analysis tools to actually verify usages of the returned values. `Nullable` is basically NOOP in SpotBugs

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 1]
```diff
@@ -165,7 +169,7 @@ public final OSProcess get(int pid) {
      *
      * Either of the parameter can be null.
      */
-    public void killAll(Process proc, Map<String, String> modelEnvVars) throws InterruptedException {
+    public void killAll(@Nullable Process proc, @Nullable Map<String, String> modelEnvVars) throws InterruptedException {
```
Comment (timja) [inline core/src/main/java/hudson/util/ProcessTree.java]:
```suggestion
    public void killAll(@CheckForNull Process proc, @CheckForNull Map<String, String> modelEnvVars) throws InterruptedException {
```

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 2]
```diff
@@ -140,7 +143,8 @@ public final OSProcess get(int pid) {
      * or null if it fails (for example, maybe the snapshot is taken after
      * this process has already finished.)
      */
-    public abstract OSProcess get(Process proc);
+    @Nullable
```
Comment (timja) [inline core/src/main/java/hudson/util/ProcessTree.java]:
```suggestion
    @CheckForNull
```

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 3]
```diff
@@ -124,13 +125,15 @@ private ProcessTree(boolean vetoesExist) {
...
      */
+    @Nullable
     public final OSProcess get(int pid) {
         return processes.get(pid);
     }
 
     /**
      * Lists all the processes in the system.
      */
+    @Nonnull
```
Comment (timja) [inline core/src/main/java/hudson/util/ProcessTree.java]:
```suggestion
    @CheckForNull
```

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 4]
```diff
@@ -124,13 +125,15 @@ private ProcessTree(boolean vetoesExist) {
     /**
      * Gets the process given a specific ID, or null if no such process exists.
      */
+    @Nullable
```
Comment (timja) [inline core/src/main/java/hudson/util/ProcessTree.java]:
```suggestion
    @CheckForNull
```

**Comment by slonopotamus** [comment]
Replaced `@Nullable` with `@CheckForNull`.

**Review by timja — APPROVED** [review]

**Comment by slonopotamus** [comment]
Two bugs detected by SpotBugs are fixed by #4231 and #4232.

**Comment by res0nance** [comment]
Requires the spotbugs fixes first so putting on hold

**Comment by slonopotamus** [comment]
@res0nance: dependent PRs got merged, so this now passes on CI and on-hold can be removed.

**Review by oleg-nenashev — APPROVED** [review]
We do not always put annotations for overrides, because SpotBugs handles it right.
But yes IDEA grumbles about it...

**Comment by oleg-nenashev** [comment]
I plan to merge it tomorrow if no negative feedback

**Pull request merged**